Tests/DiligentCoreAPITest/src/ReadOnlyDepthTest.cpp

```diff
@@ -0,0 +1,581 @@
+
+    CLEAR_DEPTH_STENCIL_FLAGS ClearFlags = CLEAR_DEPTH_FLAG;
+    if (GetDepthFormat() == TEX_FORMAT_D24_UNORM_S8_UINT || GetDepthFormat() == TEX_FORMAT_D32_FLOAT_S8X24_UINT)
+        ClearFlags |= CLEAR_STENCIL_FLAG;
+    pContext->ClearDepthStencil(m_pReadWriteDSV, ClearFlags, ClearDepth, 0, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+
+    // Draw depth-only fullscreen quad
+    DrawAttribs drawAttrs{6, DRAW_FLAG_VERIFY_ALL};
+    pContext->SetRenderTargets(0, nullptr, m_pReadWriteDSV, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    pContext->SetPipelineState(m_pDepthPSO);
+    pContext->Draw(drawAttrs);
+
+    // Draw color fullscreen quad that reads depth from the texture and performs depth test simultaneously
+    pContext->SetRenderTargets(1, &m_pRTV, m_pReadOnlyDSV, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    pContext->SetPipelineState(m_pColorPSO);
+    pContext->CommitShaderResources(m_pColorSRB, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    pContext->Draw(drawAttrs);
+
+    pSwapChain->Present();
+}
+
+TEST_P(ReadOnlyDepthTest, InRenderPass)
+{
+    auto* pEnv    = GPUTestingEnvironment::GetInstance();
+    auto* pDevice = pEnv->GetDevice();
+
+    const bool bSupported = (pDevice->GetTextureFormatInfoExt(GetDepthFormat()).BindFlags & BIND_DEPTH_STENCIL) != 0;
+    if (!bSupported)
+        GTEST_SKIP_("Depth stencil format is not supported");
+
+    auto* pSwapChain = pEnv->GetSwapChain();
+    auto* pContext   = pEnv->GetDeviceContext();
+
+    GPUTestingEnvironment::ScopedReset EnvironmentAutoReset;
+
+    // Prepare reference image
+    TakeSnapshot();
+
+    // Create render passes and framebuffers
+    InitializeDepthTexture();
+
+    RefCntAutoPtr<IRenderPass>  pDepthRenderPass;
+    RefCntAutoPtr<IFramebuffer> pDepthFramebuffer;
+    CreateDepthRenderPassAndFramebuffer(pDepthRenderPass, pDepthFramebuffer);
+
+    RefCntAutoPtr<IRenderPass>  pColorRenderPass;
+    RefCntAutoPtr<IFramebuffer> pColorFramebuffer;
+    CreateColorRenderPassAndFramebuffer(pColorRenderPass, pColorFramebuffer);
+
+    InitializePipelineStates(pDepthRenderPass, pColorRenderPass);
+    InitializeSRB();
+
+    pContext->SetVertexBuffers(0, 1, &sm_pVB, nullptr, RESOURCE_STATE_TRANSITION_MODE_TRANSITION, SET_VERTEX_BUFFERS_FLAG_RESET);
+
+    // Clear color and depth to unused colors
+    pContext->SetRenderTargets(1, &m_pRTV, m_pReadWriteDSV, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    pContext->ClearRenderTarget(m_pRTV, ClearColor, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+
+    CLEAR_DEPTH_STENCIL_FLAGS ClearFlags = CLEAR_DEPTH_FLAG;
+    if (GetDepthFormat() == TEX_FORMAT_D24_UNORM_S8_UINT || GetDepthFormat() == TEX_FORMAT_D32_FLOAT_S8X24_UINT)
+        ClearFlags |= CLEAR_STENCIL_FLAG;
+    pContext->ClearDepthStencil(m_pReadWriteDSV, ClearFlags, ClearDepth, 0, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+
+    OptimizedClearValue    ClearValues[2];
+    BeginRenderPassAttribs RPBeginAttribs;
```

Comment:
I think it's better not to reuse the same structure. It's preferable to create two variables and separate the code using a scope.